pkg/cloud/services/secretsmanager/secret.go

```diff
@@ -53,21 +53,39 @@ func (s *Service) Create(m *scope.MachineScope, data []byte) (string, int32, err
 	})
 
 	// Build the prefix.
-	prefix := path.Join(entryPrefix, string(uuid.NewUUID()))
-
+	prefix := m.GetSecretPrefix()
+	if prefix == "" {
```

Comment:
When would we have an already-set prefix?

Reply:
machine gets patched with the secret prefix when at least one secret chunk has been created, and will be set if it partially succeeded in a previous reconciliation.